internal/pkg/pluginengine/fillRefValueWithOutputs_test.go

```diff
@@ -0,0 +1,223 @@
+package pluginengine
+
+import (
+	"github.com/merico-dev/stream/pkg/util/log"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/merico-dev/stream/internal/pkg/configloader"
+	"github.com/merico-dev/stream/internal/pkg/statemanager"
+)
+
+// TestOutputsInState normal test, if the ref value is correct
+func TestOutputsInState(t *testing.T) {
+	keyReferrer := statemanager.StateKey("default-trello-github_trello-github-integ")
+	stateReferrer := statemanager.State{
+		Name:     "default-trello-github",
+		Plugin:   configloader.Plugin{Kind: "trello-github-integ", Version: "0.2.0"},
+		Options:  map[string]interface{}{"a": "value"},
+		Resource: map[string]interface{}{"a": "value"},
+	}
+
+	keystateReferee := statemanager.StateKey("default_trello")
```

Comment:
A referee is a judge in a sports game. ReferredState/Reference might be a better name.